src/Services/Azure/Kusto/KustoService.cs

```diff
@@ -0,0 +1,329 @@
+using Kusto.Data;
+using Kusto.Data.Common;
+using Kusto.Data.Net.Client;
+using Microsoft.Extensions.Options;
+using ModelContextProtocol.Server;
+
+namespace AzureMcp.Services.Azure.Kusto;
+
+public sealed class KustoService(
+    ISubscriptionService subscriptionService,
+    ICacheService cacheService) : BaseAzureService, IKustoService
+{
+    private readonly ISubscriptionService _subscriptionService = subscriptionService ?? throw new ArgumentNullException(nameof(subscriptionService));
+    private readonly ICacheService _cacheService = cacheService ?? throw new ArgumentNullException(nameof(cacheService));
+    private const string KUSTO_CLUSTERS_CACHE_KEY = "kusto_clusters";
+    private static readonly TimeSpan CACHE_DURATION = TimeSpan.FromHours(1);
+
+    private ClientRequestProperties CreateClientRequestProperties()
+    {
+        return new ClientRequestProperties
+        {
+            ClientRequestId = $"AzMcp;{Guid.NewGuid()}",
+            Application = "AzureMCP"
+        };
+    }
+
+    public async Task<List<string>> ListClusters(
+        string subscriptionId,
+        string? tenant = null,
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(subscriptionId);
+
+        // Create cache key
+        var cacheKey = string.IsNullOrEmpty(tenant)
+            ? $"{KUSTO_CLUSTERS_CACHE_KEY}_{subscriptionId}"
+            : $"{KUSTO_CLUSTERS_CACHE_KEY}_{subscriptionId}_{tenant}";
+
+        // Try to get from cache first
+        var cachedClusters = await _cacheService.GetAsync<List<string>>(cacheKey, CACHE_DURATION);
+        if (cachedClusters != null)
+        {
+            return cachedClusters;
+        }
+
+        var subscription = await _subscriptionService.GetSubscription(subscriptionId, tenant, retryPolicy);
+        var clusters = new List<string>();
+        try
+        {
+            await foreach (var cluster in subscription.GetKustoClustersAsync())
+            {
+                if (cluster?.Data?.Name != null)
+                {
+                    clusters.Add(cluster.Data.Name);
+                }
+            }
+            await _cacheService.SetAsync(cacheKey, clusters, CACHE_DURATION);
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error retrieving Kusto clusters: {ex.Message}", ex);
+        }
+        return clusters;
+    }
+
+    private static async Task<string> GetClusterUri(string subscriptionId, string clusterName, string? tenant, RetryPolicyArguments? retryPolicy)
+    {
+        var kustoService = new KustoService(null!, null!); // This should be replaced with DI or refactored for static context
+        var cluster = await kustoService.GetCluster(subscriptionId, clusterName, tenant, retryPolicy);
+        if (cluster is null)
+            throw new Exception($"Kusto cluster '{clusterName}' not found in subscription '{subscriptionId}'.");
+        var clusterUri = cluster?["clusterUri"]?.ToString();
```

Comment:
didn't see a response. did you test this? IIRC, the property is actually 'uri', but I might be wrong